### ISSUE TITLE
Window hidden behind taskbar after maximize

### ISSUE THREAD

**Issue by MrBlenny** [electron/electron#7672]
- Electron version:
  V.1.2.8
- Operating system:
  Windows 10.1

STAGE 1: I create a browserwindow using:

```
  browserWindow = new BrowserWindow({
    show: false,
    width: 1000,
    height: 600,
    minWidth: 500,
    minHeight: 500,
    frame: process.platform == 'darwin' ? true : false
  });
```

STAGE 2: From within the renderer I call:

```
    import electron from 'electron';
    const window = electron.remote.getCurrentWindow();
    window.setSize(1000, 600, true);
    window.setMinimumSize(500, 500);
    window.setResizable(false);
```

STAGE 3: Then, at a later stage I call:

```
    import electron from 'electron';
    const window = electron.remote.getCurrentWindow();
    window.setMinimumSize(500, 500);
    window.setResizable(true);
    window.maximize();
```

This final step seems to break the browser window in several ways. Windows snap stops working (dragging the window to the top/sides does nothing). And the maximise() command ignores the taskbar (window is behind the taskbar - see below).

![Taskbar above browserwindow](https://www.dropbox.com/s/e56m8gwyiw65gg6/2016-10-20%2010_27_33-.png?dl=1)

When I refresh (only STAGE 3 will run) the window is fixed. Snap works again and maximise() will not ignore the taskbar.

**Comment by MrBlenny** [comment]
Any ideas what is causing this? I'd be happy to see if I can fix it (with a few pointers).
I'm using this in production so keen to find a fix ASAP.

**Comment by kevinsawicki** [comment]
> Electron version: V.1.2.8

Can you try with Electron 1.4.4?

**Comment by MrBlenny** [comment]
Same issue in 1.4.4

**Comment by kevinsawicki** [comment]
I was able to reproduce this with the following:

```js
var w = require('electron').remote.getCurrentWindow();
w.setResizable(false);
w.setMinimumSize(500, 500);
w.maximize();
```

**Comment by kevinsawicki** [comment]
Fixed by #7765
